components/stepper/stories/template.js

```diff
@@ -10,96 +12,114 @@ import { when } from "lit/directives/when.js";
+	value = "0",
+	id = getRandomId("numberfield"),
 	customClasses = [],
 	customStyles = {},
 } = {}, context = {}) => {
-	let iconSize = "75";
-	switch (size) {
-		case "s":
-			iconSize = "50";
-			break;
-		case "l":
-			iconSize = "100";
-			break;
-		case "xl":
-			iconSize = "200";
-			break;
-		default:
-			iconSize = "75";
-	}
-
+	const { updateArgs } = context;
 	return html`
 		<div
 			class=${classMap({
 				[rootClass]: true,
-				[`${rootClass}--size${size?.toUpperCase()}`]: typeof size !== "undefined",
-				[`${rootClass}--quiet`]: isQuiet,
+				[`${rootClass}--size${size?.toUpperCase()}`]: typeof size !== "undefined" && size !== "m",
+				[`${rootClass}--sideLabel`]: labelPosition === "side",
 				"is-focused": isFocused,
 				"is-hover": isHovered,
 				"is-keyboardFocused": isKeyboardFocused,
 				"is-invalid": isInvalid,
 				"is-disabled": isDisabled,
-				"hide-stepper": hideStepper,
+				"spectrum-NumberField--hiddenStepper": hideStepper,
 				...customClasses.reduce((a, c) => ({ ...a, [c]: true }), {}),
 			})}
+			@keyup=${function(e) {
+				// Tab key was used.
+				if (e.keyCode === 9) {
+					// The element that was focused when the key was released is this textfield / input.
+					if (e.target == this || e.target?.parentNode == this) {
+						updateArgs?.({ isKeyboardFocused: true });
+						// Manually add class since updateArgs doesn't always work on the Docs page.
+						this.classList.add("is-keyboardFocused");
+					}
+				}
+			}}
+			@focusout=${function() {
+				updateArgs?.({
+					isFocused: false,
+					isKeyboardFocused: false,
+				});
+				// Manually remove class since updateArgs doesn't always work on the Docs page.
+				this.classList.remove("is-keyboardFocused");
+			}}
 			id=${ifDefined(id)}
 			style=${styleMap({
-				"--mod-actionbutton-icon-size": "10px",
 				...customStyles
 			})}
 		>
-			${Textfield({
-				size,
-				type: "number",
-				min: "-2",
-				max: "2",
-				step: "0.5",
-				value: "0",
-				isDisabled,
-				isQuiet,
-				id: id ? `${id}-input` : undefined,
-				customClasses: [`${rootClass}-textfield`],
-				customInputClasses: [`${rootClass}-input`],
-			}, context)}
-			${when(!hideStepper, () => html`
-				<span class="${rootClass}-buttons">
-					${InfieldButton({
-						size,
-						customClasses: [`${rootClass}-button`],
-						iconName: `ChevronUp${iconSize}`,
-						iconSet: "ui",
-						isDisabled,
-						isQuiet,
-						position: "top",
-						tabIndex: "-1"
-					}, context)}
-					${InfieldButton({
+			${when(displayLabel && label, () => html`
+				${FieldLabel({
+					size,
+					label: label,
+					labelPosition: labelPosition,
+					isDisabled,
+					customClasses: [`${rootClass}-fieldLabel`],
+				}, context)}
+			`)}
+			<div class="${rootClass}-inputs">
+				${Textfield({
+					size,
+					type: "number",
+					min: "-2",
+					max: "2",
+					step: "0.5",
+					value: value,
+					isInvalid,
+					isFocused,
+					isDisabled,
+					displayLabel: false,
```

Comment:
I hardcoded this `displayLabel: false` (as opposed to `displayLabel` getting passed into Textfield) so that we wouldn't get duplicate field labels. Does that make sense here, or should I refactor and just use the embedded field label that's within the textfield component? Pretty sure I can do that, with a little additional CSS edits.

Reply:
A few things to update here: 

- I can now remove the `displayLabel: false` since[ this PR](https://github.com/adobe/spectrum-css/pull/3660) to address this was merged.
- I don't mind the idea of separating the textfield input from the rest of textfield. That may encourage the "unstyled" input thing I tried in this PR. I'm happy to explore that some more.
- Right now, the `displayLabel` is a boolean for the field label because I'm pulling it from textfield's args.

Do you think I should refactor number field's field label arg to match label text and help text (getting rid of the boolean and just using a string?)

Reply:
@rise-erpelding Does this ticket (CSS-867) actually cover what we're talking about? It has some browser support steps to it, but sounds like some research into styling the inputs is involved.